model/repo_ref.go

```diff
@@ -102,3 +108,41 @@ func FindRepoRefByOwnerAndRepo(owner, repoName string) (*RepoRef, error) {
 		RepoRefRepoKey:  repoName,
 	}))
 }
+
+func (r *RepoRef) AddPermissions(creator *user.DBUser) error {
+	rm := evergreen.GetEnvironment().RoleManager()
+
+	parentScope := evergreen.AllProjectsScope
+	if !r.Restricted {
+		parentScope = evergreen.UnrestrictedProjectsScope
+	}
+
+	newScope := gimlet.Scope{
+		ID:          evergreen.GetRepoScope(r.Id),
+		Resources:   []string{r.Id},
+		Name:        r.Id,
+		Type:        evergreen.ProjectResourceType,
+		ParentScope: parentScope,
+	}
+	if err := rm.AddScope(newScope); err != nil {
+		return errors.Wrapf(err, "error adding scope for repo project '%s'", r.Id)
+	}
+
+	newRole := gimlet.Role{
+		ID:          fmt.Sprintf("admin_repo_%s", r.Id),
```

Comment:
A repo ID and project ID can't be the same can they? So I think this could be admin_project_%s. The project part was meant to identify the type of resource the role applies to, but I guess we shouldn't put that much meaning into a naming convention

Reply:
That's true so it could be admin_project_%s but I think there's no harm in the way it's currently defined, the role is different from project roles because the resources include all the projects in the repo so I'd prefer them to have distinct naming conventions.